packages/text/src/plain-text.ts

```diff
@@ -0,0 +1,98 @@
+//
+// Copyright © 2023 Hardcore Engineering Inc.
+//
+// Licensed under the Eclipse Public License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may
+// obtain a copy of the License at https://www.eclipse.org/legal/epl-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import { SaxesParser } from 'saxes'
+
+const blockTags = [
+  'address',
+  'article',
+  'aside',
+  'blockquote',
+  'canvas',
+  'dd',
+  'div',
+  'dl',
+  'dt',
+  'fieldset',
+  'figcaption',
+  'figure',
+  'footer',
+  'form',
+  'h1',
+  'h2',
+  'header',
+  'hr',
+  'li',
+  'main',
+  'nav',
+  'ol',
+  'p',
+  'pre',
+  'section',
+  'table',
+  'tfoot',
+  'ul',
+  'video'
+]
+const ELLIPSIS_CHAR = '…'
+const WHITESPACE = ' '
+
+/**
+ * @public
+ */
+export function stripTags (htmlString: string, textLimit = 0): string {
+  const parser = new SaxesParser()
```

Comment:
Why we need this? we have already facilities to parse html using tiptap and prosemirror

Reply:
Better to do parse with tiptap and render to text using it.

Reply:
You can find them here:
https://github.com/hcengineering/anticrm/blob/main/packages/text/src/html.ts

Reply:
No problem, will do this with tiptap! 

However this approach of making a plain text from HTML based on plain js sax parser implementation has its benefits. We are not using tiptap to show activity messages, only to edit and so we are rely on HTML representation. This approach is quick and doesn't require any additional assumptions.

Reply:
Markdown is hold in html tiptap/prosemirror presentation, so it is not full html syntax, only subset. we use it on server for backlinks processing. So natural will be to use it here